src/mocompiler.js

```diff
@@ -61,21 +66,31 @@ function Compiler (table = {}) {
 
     return result;
   }, {});
+}
 
-  this._table.translations = translations;
-  this._table.headers = headers;
+/**
+ * Creates a MO compiler object.
+ *
+ * @param {import('./types.js').GetTextTranslations} [table] Translation table as defined in the README
+ */
+function Compiler (table) {
+  /** @type {import('./types.js').GetTextTranslations} _table The translation table */
+  this._table = {
+    charset: undefined,
+    translations: prepareTranslations(table?.translations ?? {}),
+    headers: prepareMoHeaders(table?.headers ?? {})
+  };
 
   this._translations = [];
 
-  this._writeFunc = 'writeUInt32LE';
```

Comment:
Can this be removed?

Reply:
That assignment appears to have been established at some point, presumably intended for modification through options. However, with no implementation efforts undertaken, it remains as unused memory.

https://nodejs.org/api/buffer.html#bufwriteuint32levalue-offset

These serve as the 'placeholders' for the associated function name utilized for reading bytes from the po/mo files. Frankly, I'm uncertain whether it's worthwhile to implement the others, as file writing should adhere to a similar approach.

this made typing more complicated, what do you say I leave it or is it OK?

Reply:
I would argue that byte order should be an option of the compiler, so rather than remove the concept of changing the write function we should fix it.

If a mo file is parsed in a specific byte order it should be rewritten in the same order.

Reply:
> should be an option of the compiler

i think this requires a separate issue. won't fix in this pr